src/pyfdb/pyfdb.py

```diff
@@ -329,18 +336,76 @@ def prepare_config(c):
         # Set free function
         self.__fdb = ffi.gc(fdb[0], lib.fdb_delete_handle)
 
-    def archive(self, data, request=None) -> None:
+    @overload
+    def archive(self, data: bytes, request: Optional[Request | dict | None] = None, key: None = None) -> None: ...
+
+    @overload
+    def archive(self, data: bytes, request: None = None, key: Optional[Key | dict] = None) -> None: ...
+
+    def archive(
+        self,
+        data: bytes,
+        request: Optional[Request | dict] = None,
+        key: Optional[Key | dict] = None,
+    ) -> None:
         """Archive data into the FDB5 database
 
         Args:
-            data (bytes): bytes data to be archived
-            request (dict | None): dictionary representing the request to be associated with the data,
-                if not provided the key will be constructed from the data.
+            data: bytes data to be archived
+            request: Depending on the type, one of the following:
+                Request:
+                    Calls the fdb archive multiple API.
+                    The given data will be checked against the
+                    given request. In case of a mismatch an
```

Comment:
Should this say something like
```suggestion
                    given request. In case of a mismatch an exception will be raised.
```